labscript_devices/DummyPseudoclock/labscript_devices.py

```diff
@@ -15,24 +15,90 @@
 # and labscript. The device is a PseudoclockDevice, and can be the sole device
 # in a connection table or experiment.
 
-import labscript_utils.h5_lock
-import h5py
-from labscript import PseudoclockDevice, Pseudoclock, ClockLine
+from labscript import PseudoclockDevice, Pseudoclock, ClockLine, config, LabscriptError
+import numpy as np
 
 
 class DummyPseudoclock(PseudoclockDevice):
 
     description = 'Dummy pseudoclock'
-    clock_limit = 1e6
-    clock_resolution = 1e-6
+    clock_limit = 10e6
+    clock_resolution = 25e-9
+    clock_type = 'fast clock'
+    trigger_delay = 350e-9
+    wait_delay = 2.5e-6
+    allowed_children = None
+    max_instructions = 1e5
 
-    def __init__(self, name='dummy_pseudoclock', BLACS_connection='dummy_connection', **kwargs):
+    def __init__(
+        self, name='dummy_pseudoclock', BLACS_connection='dummy_connection', **kwargs
+    ):
         self.BLACS_connection = BLACS_connection
         PseudoclockDevice.__init__(self, name, None, None, **kwargs)
-        self.pseudoclock = Pseudoclock(self.name + '_pseudoclock', self, 'pseudoclock')
-        self.clockline = ClockLine(name='clockline', pseudoclock=self.pseudoclock, connection='dummy')
+        self._pseudoclock = Pseudoclock(
+            name=f'{name}_pseudoclock',
+            pseudoclock_device=self,
+            connection='pseudoclock',
+        )
+        self._clock_line = ClockLine(
+            name=f'{name}_clock_line',
+            pseudoclock=self.pseudoclock,
+            connection='internal',
+        )
+
+    @property
```

Comment:
Adding these `@property` decorators derived from the local attributes `self._pseudoclock` and `self._clock_line` was required (and mimics e.g. `labscript_suite.PineBlaster` and `labscript.PulseBlaster`) so that the clock line was introspected as a child device by runviewer. Otherwise no children were detected by runviewer, for reasons I cannot immediately understand.

Reply:
I can't see why that would be beyond the fact that the rest of the labscript device code in this file refers to `self.pseudoclock` and `self.clockline` rather than `self._pseudoclock` and `self._clockline`.